app/src/main/java/org/walleth/activities/MainActivity.kt

```diff
@@ -85,14 +81,7 @@ class MainActivity : AppCompatActivity(), SharedPreferences.OnSharedPreferenceCh
 
             when {
                 scanResult.isEthereumURLString() -> {
-                    val erc681 = scanResult.toERC681()
-                    if (erc681.address == null || erc681.isTokenTransfer() || (erc681.value != null && erc681.value != ZERO)) {
-                        startActivity(Intent(this, CreateTransactionActivity::class.java).apply {
-                            setData(Uri.parse(scanResult))
-                        })
-                    } else {
-                        showAddressActionChooser(erc681.address!!)
-                    }
+                    startActivity(getEthereumViewIntent(scanResult))
```

Comment:
Not sure whether that is the best solution in Kotlin. Here we have more context switching than needed. Maybe it is better to have an extension function? @ligi

Reply:
I don't see a better solution and quite like this change - is what I had in mind when opening the issue - thanks - will merge soon